aptos-move/framework/supra-framework/sources/automation_registry_state.move

```diff
@@ -0,0 +1,623 @@
+    /// Task with provided Id not found
+    const EAUTOMATION_TASK_NOT_FOUND: u64 = 4;
+    /// Gas amount does not go beyond upper cap limit
+    const EGAS_AMOUNT_UPPER: u64 = 5;
+    /// Unauthorized access: the caller is not the owner of the task
+    const EUNAUTHORIZED_TASK_OWNER: u64 = 6;
+    /// Upon new epoch entry failed to propertly calculated committed gas for the next epoch.
+    /// It is greater than current epoch committed gas.
+    const EINVALID_COMMITTED_GAS_CALCULATION: u64 = 7;
+    /// Transactoin hash that registring current task is invalid. Lenght should be 32.
+    const EINVALID_TXN_HASH: u64 = 8;
+    /// Current committed gas amount is greater than the automation gas limit.
+    const EUNACCEPTABLE_AUTOMATION_GAS_LIMIT: u64 = 9;
+    /// Trying to cancel a task which is already cancelled.
+    const EINVALID_CANCELLATION: u64 = 10;
+
+    /// The lenght of the transaction hash.
+    const TXN_HASH_LENGTH: u64 = 32;
+
+    /// Conversion factor between microseconds and second
+    const MICROSECS_CONVERSION_FACTOR: u64 = 1_000_000;
+
+    /// Constants describing task state.
+    const PENDING: u8 = 0;
+    const ACTIVE: u8 = 1;
+    const CANCELLED: u8 = 2;
+
+    /// It tracks entries both pending and completed, organized by unique indices.
+    struct AutomationRegistryState has key, store {
+        /// A collection of automation task entries that are active state.
+        tasks: EnumerableMap<u64, AutomationTaskMetaData>,
+        current_index: u64,
+        gas_committed_for_next_epoch: u64,
+        automation_gas_limit: u64,
+    }
+
+    #[event]
+    /// `AutomationTaskMetaData` represents a single automation task item, containing metadata.
+    struct AutomationTaskMetaData has copy, store, drop {
+        /// Automation task index in registry
+        id: u64,
+        /// The address of the task owner.
+        owner: address,
+        /// The function signature associated with the registry entry.
+        payload_tx: vector<u8>,
+        /// Expiry of the task, represented in a timestamp in second.
+        expiry_time: u64,
+        /// The transaction hash of the request transaction.
+        tx_hash: vector<u8>,
+        /// Max gas amount of automation task
+        max_gas_amount: u64,
+        /// Maximum gas price cap for the task
+        gas_price_cap: u64,
+        /// Registration epoch number
+        registration_epoch: u64,
+        /// Registration epoch time
+        registration_time: u64,
+        /// Flag indicating whether the task is active, canclled or pending.
+        state: u8
+    }
+
+    #[event]
+    /// Update automation gas limit event
+    struct UpdateAutomationGasLimit has drop, store {
+        automation_gas_limit: u64
+    }
+
+    #[event]
+    /// Remove automation task registry event
+    struct CanclledAutomationTask has drop, store {
+        id: u64
+    }
+
+    public fun task_expiry_time(task: &AutomationTaskMetaData): u64 {
```

Comment:
```suggestion
   #[view]
     public fun task_expiry_time(task: &AutomationTaskMetaData): u64 {
```

Reply:
this is internal function and not a view function exposed to user.